### PR TITLE
feat: parameterized cache keys for saving cache

### PR DESCRIPTION
**Pull request by ryanbourdais**
This feature should resolve the issue encountered in [issue #40](https://github.com/CircleCI-Public/android-orb/issues/40).

- Added `gradle-cache-checksum` parameter to allow custom checksums for saving and restoring caches

### CODE REVIEW

File: src/scripts/generate-cache-checksum.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 find ${PARAM_FIND_ARGS} | sort | xargs cat |
```
Comment (EricRibeiro) [inline src/scripts/generate-cache-checksum.sh:2]: Double quote the `PARAM_FIND_ARGS` variable.

File: src/scripts/write-checksum.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+if [ ${PARAM_GRADLE_CACHE_CHECKSUM} != "/tmp/gradle_cache_seed" ]; then
+    echo '${PARAM_GRADLE_CACHE_CHECKSUM}' > ${PARAM_GRADLE_CACHE_CHECKSUM}.txt
```
Comment (EricRibeiro) [inline src/scripts/write-checksum.sh:3]: As you are using single quotes, the literal string will be saved in the text file.

**Comment by felicianotech** [comment]
Some of this work is complete (it seems) and some isn't. There's a lot of conflicts. Will close this and leave the branch around for when Ryan returns.

**Comment by tjmcs** [comment]
Sounds good to me, @felicianotech

**Pull request closed without merging**